internal/pkg/cli/env_package.go

```diff
@@ -0,0 +1,298 @@
+			App:             appCfg,
+			Env:             envCfg,
+			SessionProvider: sessProvider,
+		})
+	}
+	return opts, nil
+}
+
+// Validate returns an error for any invalid optional flags.
+func (o *packageEnvOpts) Validate() error {
+	return nil
+}
+
+// Ask prompts for and validates any required flags.
+func (o *packageEnvOpts) Ask() error {
+	if o.appName == "" {
+		// This command is required to be executed under a workspace. We don't prompt for it.
+		return errNoAppInWorkspace
+	}
+
+	if _, err := o.getAppCfg(); err != nil {
+		return err
+	}
+	return o.validateOrAskEnvName()
+}
+
+// Execute prints the CloudFormation configuration for the environment.
+func (o *packageEnvOpts) Execute() error {
+	mft, err := o.environmentManifest()
+	if err != nil {
+		return err
+	}
+	principal, err := o.caller.Get()
+	if err != nil {
+		return fmt.Errorf("get caller principal identity: %v", err)
+	}
+	deployer, err := o.newEnvDeployer()
+	if err != nil {
+		return err
+	}
+
+	var urls map[string]string
+	if o.uploadAssets {
+		urls, err = deployer.UploadArtifacts()
+		if err != nil {
+			return fmt.Errorf("upload assets for environment %q: %v", o.envName, err)
+		}
+	}
+	res, err := deployer.GenerateCloudFormationTemplate(&deploy.DeployEnvironmentInput{
+		RootUserARN:         principal.RootUserARN,
+		CustomResourcesURLs: urls,
+		Manifest:            mft,
+	})
+	if err != nil {
+		return fmt.Errorf("generate CloudFormation template from environment %q manifest: %v", o.envName, err)
+	}
+	if err := o.setWriters(); err != nil {
+		return err
+	}
+	if err := o.write(o.tplWriter, res.Template); err != nil {
+		return err
+	}
+	return o.write(o.paramsWriter, res.Parameters)
+}
+
+func (o *packageEnvOpts) getAppCfg() (*config.Application, error) {
+	if o.appCfg != nil {
+		return o.appCfg, nil
+	}
+	cfg, err := o.cfgStore.GetApplication(o.appName)
+	if err != nil {
+		return nil, fmt.Errorf("get application %q configuration: %w", o.appName, err)
+	}
+	o.appCfg = cfg
+	return o.appCfg, nil
+}
+
+func (o *packageEnvOpts) getEnvCfg() (*config.Environment, error) {
+	if o.envCfg != nil {
+		return o.envCfg, nil
+	}
+	cfg, err := o.cfgStore.GetEnvironment(o.appName, o.envName)
+	if err != nil {
+		return nil, fmt.Errorf("get environment %q in application %q: %w", o.envName, o.appName, err)
+	}
+	o.envCfg = cfg
+	return o.envCfg, nil
+}
+
+func (o *packageEnvOpts) validateOrAskEnvName() error {
+	if o.envName != "" {
+		if _, err := o.getEnvCfg(); err != nil {
+			log.Errorf("It seems like environment %s is not added in application %s yet. Have you run %s?\n",
+				o.envName, o.appName, color.HighlightCode("copilot env init"))
+			return err
+		}
+		return nil
+	}
+
+	name, err := o.sel.LocalEnvironment("Select an environment manifest from your workspace", "")
+	if err != nil {
+		return fmt.Errorf("select environment: %w", err)
+	}
+	o.envName = name
+	return nil
+}
+
+func (o *packageEnvOpts) environmentManifest() (*manifest.Environment, error) {
```

Comment:
Do you think it's worthy uniting this method with the following? https://github.com/aws/copilot-cli/blob/74d034755279b4d28b3c6f4b61c9a34aa2e6513b/internal/pkg/cli/env_deploy.go#L138

Reply:
Done!